dev/n.ts

```diff
@@ -0,0 +1,175 @@
+import node from "./node";
+
+const versionArg: Fig.Arg = {
+  name: "version",
+  suggestions: [
+    {
+      name: ["latest", "current"],
+      description: "Newest official release",
+    },
+    {
+      name: ["lts"],
+      description: "Newest Long Term Support official release",
+    },
+    {
+      name: ["auto"],
+      description:
+        "Read version from file: .n-node-version, .node-version, .nvmrc, or package.json",
+    },
+    {
+      name: ["engine"],
+      description: "Read version from package.json",
+    },
+  ],
+  generators: [
+    {
+      script: "n lsr --all",
+      postProcess: function (out) {
+        const versions = out
+          .split("\n")
+          .slice(1)
+          .map((i) =>
+            i.split(".").map((_, i, a) => a.slice(0, i + 1).join("."))
+          )
+          .flat();
+        return [...new Set(versions)].map((version) => {
+          return {
+            name: [version, `v${version}`],
+            description: `Node.js ${version}`,
+          };
+        });
+      },
+    },
+  ],
+  isOptional: false,
+};
+const optionalVersionArg: Fig.Arg = {
+  ...versionArg,
+  isOptional: true,
+};
+const variadicVersionArg: Fig.Arg = {
+  ...versionArg,
+  isVariadic: true,
+};
+
+const completionSpec: Fig.Spec = {
+  name: "n",
+  description: "Node version management",
+  subcommands: [
+    {
+      name: ["i", "install"],
+      description: "Install a Node.js version",
+      args: [versionArg],
```

Comment:
nit: both for this and the others
```suggestion
      args: versionArg,
```